src/main/java/com/arcbees/analytics/server/options/ServerOptionsCallback.java

```diff
@@ -68,17 +68,17 @@ public void onCallback(String options) {
     }
 
     @Override
-    public void putBoolean(String fieldName, boolean value) {
+    public synchronized void putBoolean(String fieldName, boolean value) {
         putText(fieldName, value ? "1" : "0");
     }
 
     @Override
-    public void putNumber(String fieldName, double value) {
+    public synchronized void putNumber(String fieldName, double value) {
         putText(fieldName, value + "");
     }
 
     @Override
-    public void putText(String fieldName, String value) {
+    public synchronized void putText(String fieldName, String value) {
         if (value == null) {
             options.remove(ProtocolTranslator.getFieldName(fieldName));
```

Comment:
Oh, OK. Hardly no risk of contention then, so maybe just add a `synchronized` to `getOptions`.

Glancing at the code the other day, it seemed to me that there was some map somewhere that was in a singleton (IIRC).